### PR TITLE
quality: Docstring cleanup for pydocstyle prep

### PR DESCRIPTION
**Pull request by half-duplex**
### Description
Adds flake8-docstrings and starts cleanup for putting it in use.

First commit is only requirements, config, spacing changes, raw strings for backslashes, and punctuation. No new or meaningfully changed docstrings.

Second adds minor rephrases and reformats, like "Gets xyz" → "Get xyz".

Third adds restructures and rephrases. A couple are check-dodging ("Home directory for...") but I [disagree with D401 on properties](https://github.com/PyCQA/pydocstyle/issues/301) anyway

Relates to #1765

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

**Comment by half-duplex** [comment]
To tl;dr IRC:
- D401's heuristics and its enforcement on properties are somewhat annoying, though slightly improved in v3+
- Exirel does the most with docs and doesn't want "[this tool that] enforces only compliance, not quality"

File: sopel/config/__init__.py [hunk 1]
```diff
@@ -145,7 +148,7 @@ def __init__(self, filename, validate=True):
 
     @property
     def homedir(self):
-        """The config file's home directory.
+        """Home directory for config file.
```
Comment (Exirel) [inline sopel/config/__init__.py:151]: Ditto @ "property are not method from the reader pov".

File: sopel/irc/backends.py [hunk 2]
```diff
@@ -151,21 +152,21 @@ def initiate_connect(self, host, port, source_address):
             self.handle_close()
 
     def handle_connect(self):
-        """Called when the active opener's socket actually makes a connection."""
+        """Handle active opener's new accepted connection."""
```
Comment (Exirel) [inline sopel/irc/backends.py:155]: Lost details here.

File: sopel/loader.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def trim_docstring(doc):
 
 
 def clean_callable(func, config):
-    """Clean the callable. (compile regexes, fix docs, set defaults)
+    """Clean the callable. (compile regexes, fix docs, set defaults).
```
Comment (Exirel) [inline sopel/loader.py:37]: Is it proper grammar?

File: sopel/modules/admin.py [hunk 4]
```diff
@@ -185,9 +185,9 @@ def quit(bot, trigger):
 @plugin.priority('low')
 @plugin.example('.say #YourPants Does anyone else smell neurotoxin?')
 def say(bot, trigger):
-    """
-    Send a message to a given channel or nick. Can only be done in privmsg by
-    an admin.
+    """Send a message to a given channel or nick.
+
+    Can only be done in privmsg by an admin.
```
Comment (Exirel) [inline sopel/modules/admin.py:190]: While we are at it, let's remove the possible confusion between `PRIVMSG` and "Private message":

```suggestion
    Can only be done in a private message by an admin.
```

File: sopel/modules/admin.py [hunk 5]
```diff
@@ -205,9 +205,9 @@ def say(bot, trigger):
 @plugin.command('me')
 @plugin.priority('low')
 def me(bot, trigger):
-    """
-    Send an ACTION (/me) to a given channel or nick. Can only be done in
-    privmsg by an admin.
+    """Send an ACTION (/me) to a given channel or nick.
+
+    Can only be done in privmsg by an admin.
```
Comment (Exirel) [inline sopel/modules/admin.py:210]: Same:

```suggestion
    Can only be done in a private message by an admin.
```

File: sopel/modules/admin.py [hunk 6]
```diff
@@ -232,9 +232,7 @@ def invite_join(bot, trigger):
 @plugin.event('KICK')
 @plugin.priority('low')
 def hold_ground(bot, trigger):
-    """
-    This function monitors all kicks across all channels Sopel is in. If it
-    detects that it is the one kicked it'll automatically join that channel.
+    """Monitor KICK messages to rejoin if kicked.
```
Comment (Exirel) [inline sopel/modules/admin.py:235]:
```suggestion
    """Monitor ``KICK`` messages to rejoin if kicked.
```

File: sopel/modules/admin.py [hunk 7]
```diff
@@ -205,9 +205,9 @@ def say(bot, trigger):
 @plugin.command('me')
 @plugin.priority('low')
 def me(bot, trigger):
-    """
-    Send an ACTION (/me) to a given channel or nick. Can only be done in
-    privmsg by an admin.
+    """Send an ACTION (/me) to a given channel or nick.
```
Comment (Exirel) [inline sopel/modules/admin.py:208]:
```suggestion
    """Send an ``ACTION`` (``/me``) to a given channel or nick.
```

File: sopel/modules/adminchannel.py [hunk 8]
```diff
@@ -32,9 +32,9 @@ def default_mask(trigger):
 @plugin.require_privilege(plugin.OP, ERROR_MESSAGE_NO_PRIV)
 @plugin.command('op')
 def op(bot, trigger):
-    """
-    Command to op users in a room. If no nick is given,
-    Sopel will op the nick who sent the command
+    """Op users in a room.
```
Comment (Exirel) [inline sopel/modules/adminchannel.py:35]: Maybe we should change `room` to `channel`?

Comment (Exirel) [reply sopel/modules/adminchannel.py:35]:
```suggestion
    """Op users in a channel.
```

File: sopel/modules/adminchannel.py [hunk 9]
```diff
@@ -50,9 +50,9 @@ def op(bot, trigger):
 @plugin.require_privilege(plugin.OP, ERROR_MESSAGE_NO_PRIV)
 @plugin.command('deop')
 def deop(bot, trigger):
-    """
-    Command to deop users in a room. If no nick is given,
-    Sopel will deop the nick who sent the command
+    """Deop users in a room.
```
Comment (Exirel) [inline sopel/modules/adminchannel.py:53]:
```suggestion
    """Deop users in a channel.
```

File: sopel/modules/adminchannel.py [hunk 10]
```diff
@@ -68,9 +68,9 @@ def deop(bot, trigger):
 @plugin.require_privilege(plugin.OP, ERROR_MESSAGE_NO_PRIV)
 @plugin.command('voice')
 def voice(bot, trigger):
-    """
-    Command to voice users in a room. If no nick is given,
-    Sopel will voice the nick who sent the command
+    """Voice users in a room.
```
Comment (Exirel) [inline sopel/modules/adminchannel.py:71]:
```suggestion
    """Voice users in a channel.
```

File: sopel/modules/adminchannel.py [hunk 11]
```diff
@@ -86,9 +86,9 @@ def voice(bot, trigger):
 @plugin.require_privilege(plugin.OP, ERROR_MESSAGE_NO_PRIV)
 @plugin.command('devoice')
 def devoice(bot, trigger):
-    """
-    Command to devoice users in a room. If no nick is given,
-    Sopel will devoice the nick who sent the command
+    """Devoice users in a room.
```
Comment (Exirel) [inline sopel/modules/adminchannel.py:89]:
```suggestion
    """Devoice users in a channel.
```

**Review by Exirel — CHANGES_REQUESTED** [review]
First round of review. I still think it's too early to add a strong docstring check, and I still think these little changes don't actually improve the documentation.

**Comment by dgw** [comment]
Could revisit this now that most of the stuff to be removed in 8.0 is gone. Even so, I think we're best off with a transitional period where this check is a separate `make` target so it can report any violations in this category without failing the build (as described in https://github.com/PyCQA/flake8/issues/515#issuecomment-812812944). We can keep an eye on the status as our PRs churn and _maybe_ decide to enforce it later.

There are so many merge conflicts now that it's probably easier to add the plugin and apply fixes from scratch. Since I literally asked to use this in #1765, I'm certainly happy to be the one who goes over everything again. (Could be therapeutic after all my end-of-year stuff is done at work.)

**Comment by dgw** [comment]
We're still going to do this, but in a much less invasive way. The sane approach is to first start getting in the habit of linting everything we touch with the new rules and fixing any style violations. We can slowly work our way up to hardline enforcement at the `make quality` step in CI. But more importantly, we can avoid having one monolithic PR that tries to touch everything.

**Pull request closed without merging**